components/post_view/show_more/show_more.tsx

```diff
@@ -32,7 +34,12 @@ export default class ShowMore extends React.PureComponent<Props, State> {
 
     constructor(props: Props) {
         super(props);
-        this.maxHeight = this.props.isAttachmentText ? MAX_ATTACHMENT_TEXT_HEIGHT : MAX_POST_HEIGHT;
+        this.maxHeight = MAX_POST_HEIGHT;
+        if (this.props.attachmentTextOverflowType === 'ellipsis') {
+            this.maxHeight = 100;
+        } else if (this.props.isAttachmentText) {
+            this.maxHeight = MAX_ATTACHMENT_TEXT_HEIGHT;
+        }
```

Comment:
Maybe instead of all of this, we pass in maxHeight as a prop? I don't know how it ended up as a field to begin with, but having it potentially derived from multiple props is rather weird.

There's also the possibility of it not being updated right if the props change somehow, although I don't think that can actually happen with how we're using `ShowMore`.

Reply:
Made a maxHeight prop